js-envs/src/main/scala/org/scalajs/jsenv/nodejs/NodeJSEnv.scala

```diff
@@ -168,8 +168,11 @@ class NodeJSEnv private (
             var len = msg.length;
             var buf = new Buffer(4 + len * 2);
             buf.writeInt32BE(len, 0);
-            for (var i = 0; i < len; ++i)
-              buf.writeInt16BE(msg.charCodeAt(i), 4 + i * 2);
+            for (var i = 0; i < len; ++i) {
+              // Char code to unsigned int16
+              var c = msg.charCodeAt(i) << 16 >> 16;
+              buf.writeInt16BE(c, 4 + i * 2);
```

Comment:
Why not simply `writeUint16BE` instead of turning the chat code into signed manually?